common/unsafe/src/test/scala/org/apache/spark/unsafe/types/CollationFactorySuite.scala

```diff
@@ -409,26 +411,27 @@ class CollationFactorySuite extends AnyFunSuite with Matchers { // scalastyle:ig
 
   test("repeated and/or incompatible specifiers in collation name") {
     Seq(
-      "UTF8_LCASE_LCASE",
-      "UNICODE_CS_CS",
-      "UNICODE_CI_CI",
-      "UNICODE_CI_CS",
-      "UNICODE_CS_CI",
-      "UNICODE_AS_AS",
-      "UNICODE_AI_AI",
-      "UNICODE_AS_AI",
-      "UNICODE_AI_AS",
-      "UNICODE_AS_CS_AI",
-      "UNICODE_CS_AI_CI",
-      "UNICODE_CS_AS_CI_AI"
-    ).foreach(collationName => {
+      ("UTF8_LCASE_LCASE", "UTF8_LCASE"),
+      ("UNICODE_CS_CS", "UNICODE_CS"),
+      ("UNICODE_CI_CI", "UNICODE_CI"),
+      ("UNICODE_CI_CS", "UNICODE_CS"),
+      ("UNICODE_CS_CI", "UNICODE_CS"),
+      ("UNICODE_AS_AS", "UNICODE_AS"),
+      ("UNICODE_AI_AI", "UNICODE_AI"),
+      ("UNICODE_AS_AI", "UNICODE_AS"),
+      ("UNICODE_AI_AS", "UNICODE_AS"),
+      ("UNICODE_AS_CS_AI", "UNICODE_AS_CS"),
+      ("UNICODE_CS_AI_CI", "UNICODE_CS_AI"),
+      ("UNICODE_CS_AS_CI_AI", "UNICODE_CS_AS")
+    ).foreach { case (collationName, proposal) =>
       val error = intercept[SparkException] {
         fetchCollation(collationName)
       }
 
       assert(error.getErrorClass === "COLLATION_INVALID_NAME")
-      assert(error.getMessageParameters.asScala === Map("collationName" -> collationName))
-    })
+      assert(error.getMessageParameters.asScala === Map(
+        "collationName" -> collationName, "proposal" -> proposal))
+    }
```

Comment:
I suggest we add several more checks:
- empty locale: `_CI_AI`, `_CI_CI` and similar
- several underscores or no underscores: `locale__CI__AI`, `localeCIAI` and similar
- both trailing and leading modifiers